src/main.c

```diff
@@ -68,6 +69,36 @@ int main(int argc, char *argv[])
 		profile->rtot1 = cap->rad_ext[0];
 		profile->rtot2 = cap->rad_ext[1];
 		profile->cl = cap->length;
+		//Define output file names
+		//prf, axs and ext are just identical to input as all this info is given in there
+		cap->prf = malloc(sizeof(argv[1]));
+		if(cap->prf == NULL){
+			printf("Could not allocate cap->prf memory.\n");
+			exit(0);
+		}
+		cap->prf = argv[1];
+		cap->axs = malloc(sizeof(argv[1]));
+		if(cap->axs == NULL){
+			printf("Could not allocate cap->axs memory.\n");
+			exit(0);
+		}
+		cap->axs = argv[1];
+		cap->ext = malloc(sizeof(argv[1]));
+		if(cap->ext == NULL){
+			printf("Could not allocate cap->ext memory.\n");
+			exit(0);
+		}
+		cap->ext = argv[1];
+		//cap->out should be same as input file (FILE.inp -> FILE.out)
+		cap->out = malloc(sizeof(argv[1]));
+		if(cap->out == NULL){
+			printf("Could not allocate cap->out memory.\n");
+			exit(0);
+		}
+		char *p=strstr(argv[1],"."); //where is . in FILE.inp
+		strncpy(cap->out, argv[1], p-argv[1]); //copy argv[1] start to . in cap->out
```

Comment:
Hier gaat de memory allocation ergens de mist in. Blijkbaar zou ik iets fout doen bij cap->out allocatie, maar ditzelfde werkt wel voor cap->ext e.a. ...
Heb jij een betere/nettere manier om dit te doen, of zie jij de fout?

Reply:
sizeof werkt enkel als de lengte bij moment van compileren bekend is.

argv[1] is een pointer naar een string echter, dus zal je de lengte van die string bij runtime moeten gebruiken bij uw malloc calls. Gebruik hiervoor strlen(), maar zorg ervoor dat je altijd +1 doet voor de `\0` die op het einde moet komen...

Reply:
Alright. Lijkt nu in orde voor dit deel. Hij lijkt nog lastig te doen op einde bij f_abs in polycap_out. I'll figure it out :)